lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -196,6 +196,19 @@ def []=(key, value)
             end
           end
         end
+
+        def parse_headers(env, headers)
+          {}.tap do |result|
+            env.each do |name, value|
+              # Normalize the value, check if it's permitted.
+              tag = Datadog::Ext::HTTP::Headers.to_tag(name.to_s.sub('HTTP_', ''))
```

Comment:
let's do it the other way around; iterate over the whitelist instead (that is smaller) instead of the headers list that is in the order of thousands of elements